server/src/main/java/com/flipkart/grayskull/controllers/SecretController.java

```diff
@@ -0,0 +1,76 @@
+package com.flipkart.grayskull.controllers;
+
+import com.flipkart.grayskull.service.interfaces.SecretService;
+import com.flipkart.grayskull.models.dto.request.CreateSecretRequest;
+import com.flipkart.grayskull.models.dto.request.UpgradeSecretDataRequest;
+import com.flipkart.grayskull.models.dto.response.*;
+
+import io.swagger.v3.oas.annotations.Operation;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.validation.annotation.Validated;
+
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.Max;
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.Size;
+
+
+@RestController
+@RequestMapping("/v1/project/{projectId}/secrets")
+@RequiredArgsConstructor
+@Validated
+public class SecretController {
+
+    private final SecretService secretService;
+
+    @Operation(summary = "Lists secrets for a given project with pagination. Always returns the latest version of the secret.")
+    @GetMapping
+    @PreAuthorize("hasPermission(#projectId, T(com.flipkart.grayskull.models.authz.GrayskullActions).LIST_SECRETS)")
+    public ListSecretsResponse listSecrets(@PathVariable("projectId") @NotBlank @Size(max = 255) String projectId,
```

Comment:
`
class Response<T> {
    T data;
     String message;
    String errorCode;
}
`
we can go with standard template for all type of responses, including error responses. 

easier on client integration, especially UI.